src/java.base/unix/native/jspawnhelper/jspawnhelper.c

```diff
@@ -148,8 +165,15 @@ int main(int argc, char *argv[]) {
 #ifdef DEBUG
     jtregSimulateCrash(0, 4);
 #endif
-    r = sscanf (argv[1], "%d:%d:%d", &fdinr, &fdinw, &fdout);
-    if (r == 3 && fcntl(fdinr, F_GETFD) != -1 && fcntl(fdinw, F_GETFD) != -1) {
+    r = sscanf (argv[1], "%d:%d:%d:%d.%d.%d.%d", &fdinr, &fdinw, &fdout, &jdk_feature, &jdk_interim, &jdk_update, &jdk_patch);
+    if (r == 7 && fcntl(fdinr, F_GETFD) != -1 && fcntl(fdinw, F_GETFD) != -1) {
+        // Check that JDK version and jspawnhelper version are the same
+        if (jdk_feature != VERSION_FEATURE || jdk_interim != VERSION_INTERIM || jdk_update != VERSION_UPDATE || jdk_patch != VERSION_PATCH) {
+            fprintf(stderr, "Expected jspawnhelper for Java %d.%d.%d+%d, ", jdk_feature, jdk_interim, jdk_update, jdk_patch);
```

Comment:
This isn't correct. All of feature, interim, update and patch are always dot separated. The `+` is the separator for the build number.